docs/app-dev-context/search-attributes.md

```diff
@@ -0,0 +1,22 @@
+The typical method of retrieving a Workflow Execution is by its Workflow Id.
+However, sometimes we want to be able to retrieve one or more Executions based on other properties.
+For example, we may need to get all Executions of a certain type that have failed within a certain time range so we can start new ones with the same arguments.
+We can do this type of query with [Search Attributes](/concepts/what-is-a-search-attribute/).
```

Comment:
Ideally the term "query" should not be used at all when talking about this feature.

So one piece of feedback for this whole section is to try and come at it from the angle of "listing Workflow Executions (and bits of metadata) using List Filters".